daskernetes/core.py

```diff
@@ -55,69 +82,96 @@ class KubeCluster(object):
-        self.worker_image = worker_image
-        self.worker_labels = (worker_labels or {}).copy()
-        self.threads_per_worker = threads_per_worker
-        self.env = dict(env)
+        labels = dict(labels or {})
+        env = dict(env)
 
         # Default labels that can't be overwritten
-        self.worker_labels['org.pydata.dask/cluster-name'] = name
-        self.worker_labels['app'] = 'dask'
-        self.worker_labels['component'] = 'dask-worker'
+        labels['org.pydata.dask/cluster-name'] = name
+        labels['app'] = 'dask'
+        labels['component'] = 'dask-worker'
+        self.labels = labels
+
+        if worker_spec is None:
+            try:
+                worker_spec = config['worker']['spec']
+            except KeyError:
+                worker_spec = make_worker_spec()
+
+        worker_spec = deserialize_pod_spec(worker_spec)
+        self.worker_spec = worker_spec
+
+        self.worker_meta = client.V1ObjectMeta(
+                generate_name=name + '-',
+                labels=labels
+        )
+
+        try:
+            kubernetes.config.load_incluster_config()
+        except kubernetes.config.ConfigException:
+            kubernetes.config.load_kube_config()
+
+        self.api = client.CoreV1Api()
+
+        self.cluster = LocalCluster(ip=host or socket.gethostname(),
+                                    scheduler_port=port,
+                                    n_workers=0, **kwargs)
 
-        finalize(self, cleanup_pods, self.namespace, self.worker_labels)
+        # update env
+        env['DASK_SCHEDULER_ADDRESS'] = self.cluster.scheduler.address
+        for container in self.worker_spec.containers:
+            if container.env is None:
+                container.env = []
+            for k, v in env.items():
+                container.env.append(client.V1EnvVar(name=k, value=v))
+
+        finalize(self, cleanup_pods, self.namespace, self.labels)
 
         self._cached_widget = None
 
         if n_workers:
             self.scale_up(n_workers)
 
+    @property
+    def _worker_pod(self):
+        return client.V1Pod(metadata=self.worker_meta, spec=self.worker_spec)
+
     def _widget(self):
         """ Create IPython widget for display within a notebook """
         if self._cached_widget:
             return self._cached_widget
         import ipywidgets
         layout = ipywidgets.Layout(width='150px')
-        n_workers = ipywidgets.IntText(0, description='Workers', layout=layout)
-        threads = ipywidgets.IntText(1, description='Cores', layout=layout)
+
+        # Construct Bokeh dashboard link
+        # TODO: this currently assumes running within JupyterHub
+        # TODO: this code is entirely untested
+        link = '%sproxy/%d/status' % (os.environ['JUPYTERHUB_SERVICE_PREFIX'],
+                                      self.scheduler.services['bokeh'].port)
+        link = ipywidgets.HTML('<b>Dashboard:</b> <a href="%s" target="_blank">%s</a>' % (link, link))
```

Comment:
We still need to figure out what to do here.

Reply:
This is *very* useful in production, but it is also pretty specific.  I'm not sure yet the best way to make this externally configurable.